sync/syncer.go

```diff
@@ -222,17 +223,25 @@ func (s *Syncer) getLayerBlockIDs(index mesh.LayerID) (chan mesh.BlockID, error)
 
 func (s *Syncer) getIdsForHash(m map[string]p2p.Peer, index mesh.LayerID) (chan mesh.BlockID, error) {
 	reqCounter := 0
-	ch := make(chan []uint32)
+
+	ch := make(chan []uint32, len(m))
+	wg := sync.WaitGroup{}
 	for _, v := range m {
-		_, err := s.sendLayerIDsRequest(v, index, ch)
+		c, err := s.sendLayerIDsRequest(v, index)
 		if err != nil {
 			s.Error("could not fetch layer ", index, " block ids from peer ", v) // todo recover from this
 			return nil, err
 		} else {
 			reqCounter++
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
```

Comment:
why defer and no just put it after ? defer dosen't give you anything here

Reply:
yala sababa